src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -77,10 +85,18 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
                             featureFlag.Conditions.RequirementType));
                     }
                 }
+
+                keyValues.Add(new KeyValuePair<string, string>($"{featureFlagPath}:{FeatureManagementConstants.Status}", FeatureManagementConstants.Conditional));
             }
             else
             {
-                keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}", false.ToString()));
+                // Only explicitly set the flag to false if there are no variants to override the enabled state
+                if (featureFlag.Variants == null)
```

Comment:
I think this would need the same "Variants == null" && "Telemetry == null" logic as above.

But more importantly, I think we just remove this shortcut as well. Old FM will see no Filters and treat it as false, New FM will respect the status of Disabled